Project/js-practice/rock-paper-scissors/index.js

```diff
@@ -0,0 +1,115 @@
+let score = JSON.parse(localStorage.getItem('score')) || {
+    wins: 0,
+    losses: 0,
+    ties: 0
+  };
+  
+updateScoreElement();
+
+let isAutoPlaying = false
+let intervalID;
+function autoPlay() {
+    if (!isAutoPlaying){
+        intervalID = setInterval(() => {
+            const playerMove = pickComputerMove()
+            playGame(playerMove)
+        }, 1000);
+        isAutoPlaying = true
+    } else {
+        clearInterval(intervalID)
+        isAutoPlaying = false
+    }
+    
+}
+
+document.querySelector('.js-rock-button').addEventListener('click', () => {playGame('rock')})
+document.querySelector('.js-paper-button').addEventListener('click', () => {playGame('paper')})
+document.querySelector('.js-scissors-button').addEventListener('click', () => {playGame('scissors')})
+
+document.body.addEventListener('keydown', (event) => {
+    if (event.key === 'r') {
+        playGame('rock')
+    }
+
+    else if (event.key === 'p') {
+        playGame('paper')
+    }
+    
+    else if (event.key === 's') {
+        playGame('scissors')
+    }
+})
+
+function playGame(playerMove) {
+    const computerMove = pickComputerMove();
+  
+    let result = '';
+  
+    if (playerMove === 'scissors') {
+      if (computerMove === 'rock') {
+        result = 'You lose.';
+      } else if (computerMove === 'paper') {
+        result = 'You win.';
+      } else if (computerMove === 'scissors') {
+        result = 'Tie.';
+      }
+  
+    } else if (playerMove === 'paper') {
+      if (computerMove === 'rock') {
+        result = 'You win.';
+      } else if (computerMove === 'paper') {
+        result = 'Tie.';
+      } else if (computerMove === 'scissors') {
+        result = 'You lose.';
+      }
+      
+    } else if (playerMove === 'rock') {
+      if (computerMove === 'rock') {
+        result = 'Tie.';
+      } else if (computerMove === 'paper') {
+        result = 'You lose.';
+      } else if (computerMove === 'scissors') {
+        result = 'You win.';
+      }
+    }
+  
+    if (result === 'You win.') {
+      score.wins += 1;
+    } else if (result === 'You lose.') {
+      score.losses += 1;
+    } else if (result === 'Tie.') {
+      score.ties += 1;
+    }
+  
+    localStorage.setItem('score', JSON.stringify(score));
+  
+    updateScoreElement();
+  
+    document.querySelector('.js-result').innerHTML = result;
+  
+    document.querySelector('.js-moves').innerHTML = `You
+  <img src="images/${playerMove}-emoji.png" class="move-icon">
+  <img src="images/${computerMove}-emoji.png" class="move-icon">
+  Computer`;
+  }
+  
+function updateScoreElement() {
+    document.querySelector('.js-score')
```

Comment:
@codal-kcolaco what is querySelector and querySelectorAll used for?

Reply:
**querySelector** is used to select the first element that matches a specified CSS selector. 
**querySelectorAll** is used to select all elements that match a specified CSS selector.